src/opera/test/pge/test_metfile.py

```diff
@@ -0,0 +1,214 @@
+
+        """
+        met_file = MetFileTestCase.TEST_MET_FILE
+        met_data = MetFile(met_file)
+        self.assertIsInstance(met_data, MetFile)
+
+        # Test __setitem__ and __getitem__
+        met_data["test key"] = "test value"
+        self.assertEqual(met_data["test key"], "test value")
+        # Test write_met_file()
+        met_data.write_met_file()
+        # Copy contents of the file into a variable
+        with open(met_file, 'r') as mf:
+            lines = (mf.readlines())
+        # Test string to look for string in the dictionary
+        test_str = '  "test key": "test value",\n'
+        self.assertIn(test_str, lines)
+        # Test read_met_file()
+        met_dict = met_data.read_met_file()
+        self.assertEqual(met_data['test key'], 'test value')
+
+        # Read another line to test for an existing file
+        met_dict = "test key 2", "test value 2"
+        met_data.write_met_file()
+        with open(met_file, 'r') as mf:
+            lines = (mf.readlines())
+        # Test string to look for
+        test_str = '  "test key 2": "test value 2"\n'
+        self.assertIn(test_str, lines)
+        # Test read_met_file()
+        met_dict = met_data.read_met_file()
+        self.assertEqual(met_dict['test key 2'], 'test value 2')
+
+    def test_validate_json_file(self):
+        """
+        Instantiates pge objects
+        Verifies that the catalog metadata json file is created
+        Verifies that the schema is correct
+        Verifies that an Error is written to the log file is there is a schema error in the catalog metadata file.
+
+        """
+        # instantiate a pge object
+        runconfig_path = join(self.data_dir, 'test_base_pge_config.yaml')
+
+        pge = PgeExecutor(pge_name='BasePgeTest', runconfig_path=runconfig_path)
+
+        # Check that basic attributes were initialized
+        self.assertEqual(pge.name, "PgeExecutor")
+        self.assertEqual(pge.pge_name, "BasePgeTest")
+        self.assertEqual(pge.runconfig_path, runconfig_path)
+
+        # Kickoff execution of base PGE
+        pge.run()
+
+        # Check that runconfig and logger were instantiated as expected
+        self.assertIsInstance(pge.runconfig, RunConfig)
+        self.assertIsInstance(pge.logger, PgeLogger)
+
+        # Verify a schema check was successfully recorded in the log file
+        log_file = pge.logger.get_file_name()
+        self.assertTrue(os.path.exists(log_file))
+
+        # Open the log file, and check that the validation error details were captured
+        with open(log_file, 'r', encoding='utf-8') as log:
+            log = log.read()
+
+        # Verify the catalog metadata json file creation was logged.
+        self.assertIn("Successfully created catalog metadata json file.", log)
+
+        # Change the schema file
+        met = MetFile
+        # save the schema file
+        schema_file = met.get_schema_file_path()
+        shutil.copyfile(schema_file, 'save_file.json')
```

Comment:
I think this function needs to check for and remove `save_file.json` before its finished. See my comment here: https://github.com/nasa/opera-sds-pge/pull/38/files#r790985366

Reply:
Better yet, don't modify the schema, but create dummy metadata files to test for valid and invalid versions

Reply:
re-wrote the test to test a modified dummy metadata file.